### PR TITLE
Fix git tag checkout and other things ...

### PR DESCRIPTION
**Pull request by michaelsauter**

### CODE REVIEW

**Comment by michaelsauter** [comment]
@clemensutschig With this, I can generate the documents now. I don't like it - but anything else I tried was blocked by some Jenkins bug eventually. 

I also noticed that the TIP seems to contain wrong refs in terms of which Git ref is used for related repos. This does not use the actual value from repo.data, but seems to get what is in metadata.yml. I will look into this on Monday.

**Comment by clemensutschig** [comment]
9I Love it.. pull over my jenkins File..  and then close my pr.. whats up in the tip?

Did you also try deploy to qa?

Lets Touch Base monday to talk about the tip... in case of wip the branch is ok.. in case of Version it needs to contain the tag i believe

**Review by clemensutschig — COMMENTED** [review]
I checked the TIR ... on promotion to QA .. it lacks the tag ... 

1. the TIR (component level!) template expects `{metdata.git.targetTag}}` on repo level ... (it's only on project level in git.data.targetTag) .. so this should be a pretty easy fix - i guess in `MROPipelineUtil:prepareCheckoutRepoNamedJob` 

1. regarding the TIP - I checked the codebase, and in case of assemble mode we should have a tag there (`Project:load:  if (!getIsWorkInProgress())`)  - to be verified 

for me .. in case of WIP : no tags anywhere in case of "assemble / promotion mode" - tags

**Comment by michaelsauter** [comment]
@clemensutschig Okay, let's talk on Monday ... I'm done for the day. 

Tests are fixed now except one, however that is purely a test setup issue.

Yes I deployed to QA, documents get generated but the tag is wrong as you saw too ...

**Comment by michaelsauter** [comment]
WARNING: I also changed the target to be `master` since now the `feature/refactor-leva-docs` has been merged to `master`!

**Comment by michaelsauter** [comment]
Related are fixes in the templates: https://github.com/opendevstack/ods-document-generation-templates/commit/8135adc1a98a1ad5f14dc21fc314e1c9db0ddb15 and https://github.com/opendevstack/ods-document-generation-templates/commit/3be0ba8e70209974ad7ef89c696cd0f81c674fdf.

**Comment by michaelsauter** [comment]
Another wording fix in the templates: https://github.com/opendevstack/ods-document-generation-templates/commit/2b10cf6.

**Comment by michaelsauter** [comment]
@clemensutschig @metmajer @martsec This is ready from my point of view. Documents generate - as far as I can tell - correctly for D/WIP, D/version, Q and P.

**Comment by martsec** [comment]
> Related are fixes in the templates: [opendevstack/ods-document-generation-templates@8135adc](https://github.com/opendevstack/ods-document-generation-templates/commit/8135adc1a98a1ad5f14dc21fc314e1c9db0ddb15) and [opendevstack/ods-document-generation-templates@3be0ba8](https://github.com/opendevstack/ods-document-generation-templates/commit/3be0ba8e70209974ad7ef89c696cd0f81c674fdf).

I've added your template changes to our internal repo so we can check-it out when merging this pr.

File: vars/phaseDeploy.groovy [hunk 1]
```diff
@@ -13,18 +13,41 @@ def call(Project project, List<Set<Map>> repos) {
...
+        // In case we run the phase on an agent node, we need to make sure that
+        // the levaDocScheduler.run is executed on the master node, as it does
+        // not work on agent nodes yet.
+        if (agentPodCondition) {
+            node {
+                sh "cp -r ${standardWorkspace}/docs ${env.WORKSPACE}/docs"
+                levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.POST_EXECUTE_REPO, repo, repo.data)
+            }
+        } else {
+            levaDocScheduler.run(phase, MROPipelineUtil.PipelinePhaseLifecycleStage.PRE_EXECUTE_REPO, repo)
```
Comment (martsec) [inline vars/phaseDeploy.groovy]: Shouldn't this be `POST_EXECUTE_REPO` and contain `repo.data` ? 
And this agent pod condition here and in line 23 looks like a true duplication, right? Could it be extracted-out into a method named something like `runScedulerOnMasterNode`

File: vars/phaseDeploy.groovy [hunk 2]
```diff
@@ -13,18 +13,41 @@ def call(Project project, List<Set<Map>> repos) {
 
     def phase = MROPipelineUtil.PipelinePhases.DEPLOY
 
+    def standardWorkspace = env.WORKSPACE
+    def agentPodCondition = project.isPromotionMode
```
Comment (martsec) [inline vars/phaseDeploy.groovy:19]: It is not clear to me the naming of this boolean variable. Maybe renaming it to `isAgentPod` could help.

**Review by martsec — CHANGES_REQUESTED** [review]
Some minor comments. But I think there is one issue in `phaseDeploy` file line 43

File: vars/phaseDeploy.groovy (hunk 1 above)
Comment (clemensutschig) [reply vars/phaseDeploy.groovy]: @michaelsauter  - that's a true bug ...  :)

Comment (michaelsauter) [reply vars/phaseDeploy.groovy]: @martsec Great catch! Resolved in 0f5bc4b.

File: vars/phaseDeploy.groovy (hunk 2 above)
Comment (michaelsauter) [reply vars/phaseDeploy.groovy:19]: Hmm, let's leave it like it is for now - the method arg is named the same way.

In any case - this is just a hack. We should work on moving everything to the slave after the release.

**Review by martsec — DISMISSED** [review]

**Review by metmajer — DISMISSED** [review]

**Comment by metmajer** [comment]
@michaelsauter can't wait to have this in master. Can you please squash and merge asap (tomorrow morning)? Cheers!

**Review by martsec — DISMISSED** [review]

**Comment by clemensutschig** [comment]
@michaelsauter  - found one more - please do NOT merge ... 

in case a quickstarter of type `ods-test` is added ... looks like the branch / tags are created in assembly mode, but appearently not, as the promotion fails

assemble:
```
spock:::: 
To ...... odsst-release-m.git
 * [new branch]      release/6 -> release/6
 * [new tag]         v6-UNDEFINED-0-D -> v6-UNDEFINED-0-D
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
```
looks like the component is treated wrongly ...  I looked at the logic below ..

https://github.com/opendevstack/ods-mro-jenkins-shared-library/blob/fix/git-tag-checkout/src/org/ods/util/MROPipelineUtil.groovy#L471-L477

and I don't get it ... we could tag and push in phase test ... maybe the directory we are in is wrong? 

Found it - we don't wrap the steps with `dir` .. same for type shell script..

**Pull request merged**